python/ray/experimental/data/read_api.py

```diff
@@ -17,12 +17,30 @@
 def autoinit_ray(f):
     def wrapped(*a, **kw):
         if not ray.is_initialized():
-            ray.client.connect()
+            ray.client().connect()
         return f(*a, **kw)
 
     return wrapped
 
 
+def _parse_paths(paths: Union[str, List[str]]
+                 ) -> Tuple["pyarrow.fs.FileSystem", Union[str, List[str]]]:
+    from pyarrow import fs
+    if isinstance(paths, str):
+        return fs.FileSystem.from_uri(paths)
```

Comment:
I think that this (and the below) `FileSystem.from_uri()` call will fail on relative local path strings, which might be an issue. I think that's why pyarrow will ping the local filesystem first to see if the path is a local relative path before it tries to use `FileSystem.from_uri()`: https://github.com/apache/arrow/blob/1430c93f68960e10a50d27f465eb174e76ac06b2/python/pyarrow/fs.py#L158-L168

Reply:
Good catch! I think we should do similar things here.